### PR TITLE
chore: add event logger to reports/alerts CRUD

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY

Previously we did not have any tracking for Report CRUD actions, this adds that. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by lyndsiWilliams — APPROVED** [review]

File: superset/reports/api.py [hunk 1]
```diff
@@ -227,8 +228,12 @@ def ensure_alert_reports_enabled(self) -> Optional[Response]:
     @expose("/<int:pk>", methods=["DELETE"])
     @protect()
     @safe
-    @statsd_metrics
     @permission_name("delete")
+    @statsd_metrics
+    @event_logger.log_this_with_context(
+        action=lambda self, *args, **kwargs: f"{self.__class__.__name__}.data",
```
Comment (hughhhh) [inline superset/reports/api.py:234]: `action=lambda self, *args, **kwargs: f"{self.__class__.__name__}.data",`

File: superset/reports/api.py [hunk 2]
```diff
@@ -416,6 +429,10 @@ def put(self, pk: int) -> Response:
     @safe
     @statsd_metrics
     @rison(get_delete_ids_schema)
+    @event_logger.log_this_with_context(
+        action=lambda self, *args, **kwargs: f"{self.__class__.__name__}.delete",
```
Comment (hughhhh) [inline superset/reports/api.py:433]: `.delete_bulk`

File: superset/reports/api.py [hunk 3]
```diff
@@ -281,6 +286,10 @@ def delete(self, pk: int) -> Response:
     @statsd_metrics
     @permission_name("post")
     @requires_json
+    @event_logger.log_this_with_context(
+        action=lambda self, *args, **kwargs: f"{self.__class__.__name__}.create",
```
Comment (hughhhh) [inline superset/reports/api.py:290]: post

File: superset/reports/api.py [hunk 4]
```diff
@@ -344,6 +353,10 @@ def post(self) -> Response:
     @statsd_metrics
     @permission_name("put")
     @requires_json
+    @event_logger.log_this_with_context(
+        action=lambda self, *args, **kwargs: f"{self.__class__.__name__}.modify",
```
Comment (hughhhh) [inline superset/reports/api.py:357]: post

**Review by hughhhh — APPROVED** [review]

File: superset/reports/api.py (hunk 4 above)
Comment (AAfghahi) [reply superset/reports/api.py]: put?

**Review by vincentkoc — APPROVED** [review]

**Pull request merged**